DisposableAnnotations/DisposeAttribute.cs

```diff
@@ -0,0 +1,12 @@
+namespace DisposableAnnotations
+{
+    using System;
+
+    /// <summary>
+    /// The return value should be disposed by caller.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.ReturnValue, AllowMultiple = false, Inherited = true)]
+    public class DisposeAttribute : Attribute
```

Comment:
Just spaghetti: `[OwnedByCaller]`?

Reply:
I think having `Disposable` somewhere in the name is good.